src/firstboot/firstboot.c

```diff
@@ -285,6 +288,83 @@ static int process_locale(void) {
         return 0;
 }
 
+static int prompt_keymap(void) {
+        _cleanup_strv_free_ char **kmaps = NULL;
+        int r;
+
+        if (arg_keymap)
+                return 0;
+
+        if (!arg_prompt_keymap)
+                return 0;
+
+        r = get_keymaps(&kmaps);
+        if (r < 0)
+                return log_error_errno(r, "Cannot query keymap list: %m");
+
+        print_welcome();
+
+        printf("\nAvailable keymaps:\n\n");
+        r = show_menu(kmaps, 3, 22, 60);
+        if (r < 0)
+                return r;
+
+        putchar('\n');
+
+        r = prompt_loop("Please enter system keymap name or number", kmaps, keymap_is_valid, &arg_keymap);
+        if (r < 0)
+                return r;
+
+        if (isempty(arg_keymap))
+                return 0;
+
+        return 0;
+}
+
+static int process_keymap(void) {
+        const char *etc_vconsoleconf;
+        char* keymaps[3];
```

Comment:
you only ever add a single entry to this array, so this should be sized 2, not 3